### PR TITLE
ObjBlockStop OK and documented

### PR DESCRIPTION
**Pull request by nckwntzl**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 1]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:37]:
```suggestion
    ObjBlockstop* this = THIS;
    
    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
        Actor_MarkForDeath(&this->actor);
    }
    this->actionFunc = ObjBlockstop_CheckCollision;
```
We like to keep a space between variables and the body of functions as part of the coding style.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 2]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor;
+    tempActor = globalCtx->actorCtx.actorLists[6].first;
+    while (tempActor) {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:43]:
```suggestion
void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
    
    while (tempActor) {
```

Could this be put all on the same line?
Also, you can use the enum value for the `6`.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 3]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+                ActorCutscene_SetIntentToPlay((s16)this->actor.cutscene);
+                this->actionFunc = ObjBlockstop_TryPlayCutscene;
+            }
+        }
+        tempActor = tempActor->next;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:59]:
```suggestion
void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
```

Any particular reason this isn't `ObjBlockstop_TryPlayCutscene(ObjBlockstop* this, GlobalContext* globalCtx)`?
With that fixed, you should be able to clean up a lot of the casts in this function as well.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 4]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+        return;
+    }
+    ActorCutscene_SetIntentToPlay((s16)arg0->actor.cutscene);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Update.s")
+void ObjBlockstop_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = (ObjBlockstop*)thisx;
+    this->actionFunc(this, globalCtx);
+}
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:74]:
```suggestion
void ObjBlockstop_Update(Actor* thisx, GlobalContext* globalCtx) {
    ObjBlockstop* this = (ObjBlockstop*)thisx;
    
    this->actionFunc(this, globalCtx);
}
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 5]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+                this->actionFunc = ObjBlockstop_TryPlayCutscene;
+            }
+        }
+        tempActor = tempActor->next;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
+    if (ActorCutscene_GetCanPlayNext((s16)arg0->actor.cutscene) != 0) {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:60]:
```suggestion
    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 6]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+        Flags_SetSwitch(arg1, (s32)arg0->actor.params);
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:61]:
```suggestion
        Flags_SetSwitch(globalCtx, this->actor.params);
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 7]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+        if (ActorCutscene_GetLength((s16)arg0->actor.cutscene) != -1) {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:62]:
```suggestion
        if (ActorCutscene_GetLength(this->actor.cutscene) != -1) {
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 8]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+            ActorCutscene_StartAndSetUnkLinkFields((s16)arg0->actor.cutscene, &arg0->actor);
+        }
+        Actor_MarkForDeath(&arg0->actor);
+        return;
+    }
+    ActorCutscene_SetIntentToPlay((s16)arg0->actor.cutscene);
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:68]:
```suggestion
    ActorCutscene_SetIntentToPlay(this->actor.cutscene);
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 9]
```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
+    if (ActorCutscene_GetCanPlayNext((s16)arg0->actor.cutscene) != 0) {
+        Flags_SetSwitch(arg1, (s32)arg0->actor.params);
+        if (ActorCutscene_GetLength((s16)arg0->actor.cutscene) != -1) {
+            ActorCutscene_StartAndSetUnkLinkFields((s16)arg0->actor.cutscene, &arg0->actor);
+        }
+        Actor_MarkForDeath(&arg0->actor);
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:65]:
```suggestion
        Actor_MarkForDeath(&this->actor);
```

**Review by jpburnett — COMMENTED** [review]
Good work on your first contribution! I have a couple of suggestions to clean this up and fit the coding standard we like to follow. Feel free ask any questions or stop by the discord server if there are any questions!

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 10]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:35]:
```suggestion
    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
```
Arithmetic casts in functions are almost always not needed.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 11]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+    while (tempActor) {
+        if ((tempActor->id == 0x7A) && // check if oshihiki (push block)
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:45]: You can use the ActorId enum here
```suggestion
        if ((tempActor->id == ACTOR_OBJ_OSHIHIKI) &&
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 12]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+            (fabsf(tempActor->world.pos.x - this->actor.world.pos.x) < 20.0f) &&
+            (fabsf(tempActor->world.pos.z - this->actor.world.pos.z) < 20.0f) &&
+            (fabsf(tempActor->world.pos.y - this->actor.world.pos.y) < 20.0f)) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Update.s")
+            s32 params = OBJOSHIHIKI_GET_F(tempActor);
+            if (params < 3) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:51]: Is this temp required?

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 13]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
...
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+    while (tempActor) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:44]: We compare pointers to NULL where possible
```suggestion
    while (tempActor != NULL) {
```

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 14]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor = globalCtx->actorCtx.actorLists[ACTORCAT_PROP].first;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:42]: Just `actor` would be fine, or `prop` to be more specific.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c [hunk 15]
```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:41]: We have a system called `Collision`, so I think we should use a different name for this function. `ObjBlockstop_CheckForBlock` or something similar?

**Review by EllipticEllipsis — COMMENTED** [review]
Good job on this, just a few minor things.

**Comment by EllipticEllipsis** [comment]
Also, please resolve finished conversations where the comments have been addressed, it makes it easier to see what's left.

File: src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c (hunk 12 above)
Comment (nckwntzl) [reply src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c:51]: Yes, this got the function to match

**Review by EllipticEllipsis — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
lgtm 👍

**Review by Kenix3 — APPROVED** [review]
Looks good to me!

**Pull request merged**